cpp/include/cudf/groupby.hpp

```diff
@@ -162,7 +168,16 @@ class groupby {
       std::vector<aggregation_request> const& requests,
       rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
 
- private:
+  groupby_groups groups(
+    rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource(),
+    cudaStream_t stream = 0);
+
+  groupby_groups groups(
+    cudf::table_view values,
```

Comment:
Instead of providing two explicit overloads, maybe just default the argument of `values` and check if it is empty. 
```suggestion
    cudf::table_view values = {},
```

Reply:
Done